File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/post/usecases/ReactToPostUseCase.kt
package com.adqmobile.socialnetwork.domain.post.usecases

import com.adqmobile.socialnetwork.domain.post.PostRepository
import com.adqmobile.socialnetwork.domain.post.Reaction

class ReactToPostUseCase(private val postRepository: PostRepository) {

    suspend fun run(reaction: Reaction) {
        postRepository.reactToPost(reaction)
    }
}

File: react-native/domain/post/usecases/ReactToPostUseCase.ts
import Post from "../Post";
import PostRepository from "../PostRepository";
import Reaction from "../Reaction";

export default class ReactToPostUseCase {
    postRepository: PostRepository;

    constructor(postRepository: PostRepository) {
        this.postRepository = postRepository;
    }

    async run(reaction: Reaction): Promise<Post> {
        return await this.postRepository.reactToPost(reaction);
    }
}